src/ucp/core/ucp_am.c

```diff
@@ -1347,11 +1387,20 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_am_long_middle_handler,
     ucp_am_mid_hdr_t *mid_hdr  = am_data;
     ucp_ep_h ep                = ucp_worker_get_ep_by_id(worker,
                                                          mid_hdr->ep_id);
-    ucp_ep_ext_proto_t *ep_ext = ucp_ep_ext_proto(ep);
     uint64_t msg_id            = mid_hdr->msg_id;
     ucp_recv_desc_t *mid_rdesc = NULL, *first_rdesc = NULL;
+    ucp_ep_ext_proto_t *ep_ext;
     ucs_status_t status;
 
+    if (ucs_unlikely(!ucp_am_ep_is_valid(ep))) {
```

Comment:
btw, what if this is the last fragment received?
I guess, we could process it and report to a user, if it is not a REPLY proto.

Reply:
imo, this would make code more complicated (need to process the message before dropping) while does not provide valuable benefit